File: b) stack dynamic.java
import java.util.Scanner;

public class StackDynamic {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter array size: ");
        int size = scanner.nextInt();
        
        // Stack dynamic array - size determined at runtime
        int[] arr = new int[size];
        
        System.out.println("Enter " + size + " elements:");
        for(int i = 0; i < size; i++) {
            arr[i] = scanner.nextInt();
        }
        
        System.out.println("Stack dynamic array in Java:");
        for(int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
        
        scanner.close();
    }
}

File: C++/b) stack dynamic.cpp
#include <iostream>
using namespace std;

int main() {
    int size;
    cout << "Enter array size: ";
    cin >> size;

    // Stack dynamic array - size determined at runtime
    int arr[size];

    cout << "Enter " << size << " elements:" << endl;
    for(int i = 0; i < size; i++) {
        cin >> arr[i];
    }

    cout << "Stack dynamic array in C++:" << endl;
    for(int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;

    return 0;
}
